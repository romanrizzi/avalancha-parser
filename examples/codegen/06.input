
fun sum
  Zero   , y -> y
  Suc(x) , y -> Suc(sum(x, y))

fun mul
  Zero   , _ -> Zero
  Suc(x) , y -> sum(y, mul(x, y))

print mul(
  Suc(Suc(Suc(Zero))),
  Suc(Suc(Suc(Suc(Zero))))
)

