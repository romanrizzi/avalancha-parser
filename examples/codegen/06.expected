Suc(Suc(Suc(Suc(Suc(Suc(Suc(Suc(Suc(Suc(Suc(Suc(Zero))))))))))))
